xarray/convert.py

```
@@ -49,3 +64,130 @@ def set_cdms2_attrs(var, attrs):
+    """ Return a dictionary of attrs when given a Iris object
+    """
+    attrs = {'standard_name': obj.standard_name,
+             'long_name': obj.long_name}
+    if obj.units.calendar:
+        attrs['calendar'] = obj.units.calendar
+    if obj.units.origin != '1':
+        attrs['units'] = obj.units.origin
+    attrs.update(obj.attributes)
+    return dict((k, v) for k, v in attrs.items() if v is not None)
+
+
+def _iris_cell_methods_to_str(cell_methods_obj):
+    """ Converts a Iris cell methods into a string
+    """
+    cell_methods = []
+    for cell_method in cell_methods_obj:
+        names = ''.join(['{}: '.format(n) for n in cell_method.coord_names])
+        intervals = ' '.join(['interval: {}'.format(interval)
+                              for interval in cell_method.intervals])
+        comments = ' '.join(['comment: {}'.format(comment)
+                             for comment in cell_method.comments])
+        extra = ' '.join([intervals, comments]).strip()
+        if extra:
+            extra = ' ({})'.format(extra)
+        cell_methods.append(names + cell_method.method + extra)
+    return ' '.join(cell_methods)
+
+
+def from_iris(cube):
+    """ Convert a Iris cube into an DataArray
+    """
+    import iris.exceptions
+    name = cube.var_name
+    dims = []
+    for i in xrange(cube.ndim):
+        try:
+            dim_coord = cube.coord(dim_coords=True, dimensions=(i,))
+            dims.append(dim_coord.var_name)
+        except iris.exceptions.CoordinateNotFoundError:
+            dims.append("dim_{}".format(i))
+
+    # dims = [dim.var_name for dim in cube.dim_coords]
+    # if not dims:
+    #     dims = ["dim{}".format(i) for i in range(cube.data.ndim)]
+    coords = OrderedDict()
+
+    for coord in cube.coords():
+        coord_attrs = _iris_obj_to_attrs(coord)
+        coord_dims = [dims[i] for i in cube.coord_dims(coord)]
+        if coord_dims:
+            coords[coord.var_name] = (coord_dims, coord.points, coord_attrs)
```

I think it's also possible to `coord.var_name` be None. We should probably raise an error in that case.